app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  has_many :likes, foreign_key: :post_id
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments, foreign_key: :post_id
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] [OPTIONAL] Kindly use `after_save` to call `update_posts_counter` every time a new post has been saved to the database, then make the `update_posts_counter` method private. You can read more about `after_save` from [here](https://api.rubyonrails.org/v7.0.4/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-after_save:~:text=after_save(*args%2C%20%26block)).